### PR TITLE
Accessibility

### PR DESCRIPTION
**Pull request by nclayt0n**
- Added sass-color-helpers package.
   - copied it's scss folder into our relative scss folder to make use of their functions. 
   - Using `ch-best-color-contrast()` in our color-contrast function.
- Updated collapse button to be more accessible.
- Updated Nav links to be mapped and added aria-labels to icons.
- Added color var root, and updated content page styles to be more dynamic.
- Added an accessibility page to be add accessibility content to.

### CODE REVIEW

File: src/scss/base/_typography.scss [hunk 1]
```diff
@@ -149,6 +149,10 @@ blockquote {
     top: 2px;
 }
 
+.capitalize {
+    text-transform: capitalize !important;
+}
+
 // Text align
 .txt-center {
     text-align: center !important;
```
Comment (nsmedira) [inline src/scss/base/_typography.scss:158]: should we add text transformation examples/docs to the Content page (or some other page)?

**Review by nsmedira — APPROVED** [review]
reviewed and approved 2022-09-06 at 16:00 ET

File: src/pages/Content.jsx [hunk 2]
```diff
@@ -94,30 +94,42 @@ const ContentPage = () => {
...
-                            <div key={`colors-${color}`} className="col-6 col-md-2 txt-center">
-                                <div className={`bg-${color} p-4 mb-4`}>
-                                    <code>${color}</code>
-                                    <br />
-                                    <code>{hex}</code>
+                        {colors.map(([color]) => {
+                            let hex = rootStyles?.getPropertyValue(`--${color}`);
+                            if (!hex) {
+                                return <p> {color} not found</p>;
+                            }
```
[the remark this answers is not in the document]
Comment (nclayt0n) [reply src/pages/Content.jsx]: https://github.com/FindawayWorld/Solon/pull/82/commits/35e08c24ce54697808c3dc71cac2cdb9b721b5ad

File: src/pages/Accessibility.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    <Code
+                        lang="html"
+                        code={`<button>
+    <VisuallyHidden>Expand Sidebar</VisuallyHidden>
+    <FaChevronLeft aria-hidden focusable="false" />
+</button>`}
+                    />
+                </div>
+                <p>
+                    <strong>Summary:</strong> Both examples above have the aria-hidden attribute - defaulting to true
```
[the remark this answers is not in the document]
Comment (nclayt0n) [reply src/pages/Accessibility.jsx:31]: I dont understand the comment

File: src/scss/base/_typography.scss (hunk 1 above)
Comment (nclayt0n) [reply src/scss/base/_typography.scss:158]: https://github.com/FindawayWorld/Solon/pull/82/commits/7e678d467e56c3752b80bbeda36c45c3f655c117

File: src/pages/Accessibility.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nsmedira) [reply src/pages/Accessibility.jsx:31]: e.g.

```tsx
<p>Both examples above have the <code>aria-hidden</code> attribute - defaulting to <code>true</code></p>
```

**Review by nsmedira — APPROVED** [review]
reviewed and approved 2022-09-07 at 09:54 ET

File: src/pages/Accessibility.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                    when present. Since they are wrapped in a button that has either, an aria-label or text within the
+                    VisuallyHidden, the screen reader will read that the element is a button and read the button's
+                    purpose, the label. Therefore the screen reader never needs to focus on the icon, which is why we
+                    set aria-hidden to true and focusable false.
+                </p>
+                <p>
+                    It's good practice to include <code>focusable="false"</code> with aria-hidden because the attribute
+                    <code> aria-hidden="true"</code> hides an element and all its children from assistive technologies,
+                    but users can still use the keyboard to navigate to any focusable child elements, but their content
+                    is inaccessible to people who use assistive technologies.{' '}
```
Comment (skerr924) [inline src/pages/Accessibility.jsx:41]: extra `{' '}`

File: src/pages/Content.jsx [hunk 5]
```diff
@@ -94,30 +94,42 @@ const ContentPage = () => {
...
                 <div className="swatch-list">
                     <div className="row">
-                        {brandColors.map(([color, hex]) => (
-                            <div key={`brand-${color}`} className="col-6 col-md-2 txt-center">
-                                <div className={`bg-${color} p-4 mb-4`}>
-                                    <code>${color}</code>
-                                    <br />
-                                    <code>{hex}</code>
+                        {brandColors.map(([color, name]) => {
+                            let hex = rootStyles?.getPropertyValue(`--${color}`);
```
Comment (skerr924) [inline src/pages/Content.jsx:119]: `const` if possible

File: src/pages/Content.jsx [hunk 6]
```diff
@@ -227,7 +240,15 @@ $font-family-monospace: SFMono-Regular, Menlo, Monaco, Consolas, 'Liberation Mon
                     </p>
                 ))}
             </section>
+            <section id="text-tranform" className="mb-5">
+                <h2>Text Transform</h2>
+                <hr />
+                <p>
+                    Text Tranform Capitalize can be applied using <code>{`.capitalize`}</code>
```
Comment (skerr924) [inline src/pages/Content.jsx:247]: `Tranform`

**Review by skerr924 — COMMENTED** [review]
reviewed with minor comments 9/7 10:01 EDT

File: src/pages/Content.jsx (hunk 6 above)
Comment (nclayt0n) [reply src/pages/Content.jsx]: weird my spell check extension didnt catch it.
https://github.com/FindawayWorld/Solon/pull/82/commits/d2b5d98c3cf4f21f19f1b50cedfe9b044b9ac6a3

File: src/pages/Content.jsx [hunk 7]
```diff
@@ -94,30 +94,42 @@ const ContentPage = () => {
...
                 <div className="swatch-list mb-3">
                     <div className="row">
-                        {colors.map(([color, hex]) => (
-                            <div key={`colors-${color}`} className="col-6 col-md-2 txt-center">
-                                <div className={`bg-${color} p-4 mb-4`}>
-                                    <code>${color}</code>
-                                    <br />
-                                    <code>{hex}</code>
+                        {colors.map(([color]) => {
+                            let hex = rootStyles?.getPropertyValue(`--${color}`);
```
[the remark this answers is not in the document]
Comment (nclayt0n) [reply src/pages/Content.jsx]: https://github.com/FindawayWorld/Solon/pull/82/commits/8a6e8f250e8e6166964e496ba2f3060fc7d99141

File: src/pages/Accessibility.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nclayt0n) [reply src/pages/Accessibility.jsx]: I hope I understood what your comment was driving at - if not lmk.
https://github.com/FindawayWorld/Solon/pull/82/commits/afd9bf03140a28ebb0ba591bf3ca6c3e1438ef4c

File: src/pages/Accessibility.jsx [hunk 8]
```diff
@@ -3,45 +3,64 @@ import Code from '../components/Code';
...
+                <p>The purpose of the element visually must also be represented audibly.</p>
+            </div>
+            <section className="mb-12">
+                <h3>Aria-hidden and Focusable Attributes</h3>
+                <hr />
+                <div className="flash">
+                    <p>
+                        It's good practice to include <code>focusable="false"</code> with <code>aria-hidden</code>{' '}
+                        because the attribute
+                        <code> aria-hidden="true"</code> hides an element and all its children from assistive
```
Comment (nsmedira) [inline src/pages/Accessibility.jsx:19]: extra space within `<code>`?

File: src/pages/Accessibility.jsx [hunk 9]
```diff
@@ -3,45 +3,64 @@ import Code from '../components/Code';
...
     <VisuallyHidden>Expand Sidebar</VisuallyHidden>
     <FaChevronLeft aria-hidden focusable="false" />
 </button>`}
-                    />
+                        />
+                    </div>
+                    <p>
+                        <strong>Summary:</strong> Both examples above have the <code>aria-hidden</code> attribute -
+                        defaulting to <code>true</code> when present. Since they are wrapped in a button that has
+                        either, an aria-label or text within the <code>VisuallyHidden</code>, the screen reader will
```
Comment (nsmedira) [inline src/pages/Accessibility.jsx:44]: wrap "aria-label" in `<code>`?

**Review by nsmedira — APPROVED** [review]
reviewed and approved 2022-09-07 at 14:33 ET

File: src/pages/Accessibility.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nsmedira) [reply src/pages/Accessibility.jsx]: yes, we're on same page 👍

File: src/pages/Accessibility.jsx (hunk 9 above)
Comment (nclayt0n) [reply src/pages/Accessibility.jsx]: https://github.com/FindawayWorld/Solon/pull/82/commits/ec28c233eb5466747c3ef9f22d2ba1b105a52ee5

**Review by nsmedira — APPROVED** [review]
reviewed and approved 2022-09-09 at 11:43 ET

**Pull request merged**